Source/WebKit/UIProcess/API/glib/WebKitImage.cpp

```diff
@@ -0,0 +1,381 @@
+
+    /**
+     * WebKitImage:height:
+     *
+     * The image height
+     */
+    sObjProperties[PROP_HEIGHT] =
+    g_param_spec_int(
+        "height",
+        nullptr, nullptr,
+        0, G_MAXINT, 0,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    /**
+     * WebKitImage:stride:
+     *
+     * The image stride
+     */
+    sObjProperties[PROP_STRIDE] =
+    g_param_spec_uint(
+        "stride",
+        nullptr, nullptr,
+        0, G_MAXUINT, 0,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    /**
+     * WebKitImage:type:
+     *
+     * The image type
+     */
+    sObjProperties[PROP_TYPE] =
+    g_param_spec_string(
+        "type",
+        nullptr, nullptr,
+        nullptr,
+        static_cast<GParamFlags>(WEBKIT_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY));
+
+    g_object_class_install_properties(objectClass, N_PROPERTIES, sObjProperties.data());
+}
+
+WebKitImage* webkitImageNew(int width, int height, guint stride, const CString& type, WebKitImageCreateBytesFunction&& createBytesFn)
+{
+    g_return_val_if_fail(createBytesFn, nullptr);
+
+    auto* image = WEBKIT_IMAGE(g_object_new(WEBKIT_TYPE_IMAGE,
+        "width", width,
+        "height", height,
+        "stride", stride,
+        "type", type.data(),
+        nullptr));
+
+    image->priv->createBytesFn = WTFMove(createBytesFn);
+
+    return image;
+}
+
+/**
+ * webkit_image_get_width:
+ * @image: a #WebKitImage
+ *
+ * Get the @image width
+ *
+ * Returns: the image width
+ */
+int webkit_image_get_width(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), 0);
+    return image->priv->width;
+}
+
+/**
+ * webkit_image_get_height:
+ * @image: a #WebKitImage
+ *
+ * Get the @image height
+ *
+ * Returns: the image height
+ */
+int webkit_image_get_height(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), 0);
+    return image->priv->height;
+}
+
+/**
+ * webkit_image_get_stride:
+ * @image: a #WebKitImage
+ *
+ * Get the @image stride
+ *
+ * Returns: the image stride
+ */
+guint webkit_image_get_stride(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), 0);
+    return image->priv->stride;
+}
+
+/**
+ * webkit_image_get_image_type:
+ * @image: a #WebKitImage
+ *
+ * Get the @image type
+ *
+ * Returns: (nullable): the image type or %NULL
+ */
+const gchar* webkit_image_get_image_type(WebKitImage *image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), nullptr);
+
+    return image->priv->type.data();
+}
+
+/**
+ * webkit_image_as_rgba_bytes:
+ * @image: a #WebKitImage
+ *
+ * Get the @image bytes in their native format
+ *
+ * Returns: (transfer none): a #GBytes
+ */
+GBytes* webkit_image_as_rgba_bytes(WebKitImage* image)
+{
+    g_return_val_if_fail(WEBKIT_IS_IMAGE(image), nullptr);
+
+    if (!image->priv->bytes)
+        image->priv->bytes = image->priv->createBytesFn(image);
+    ASSERT(image->priv->bytes);
```

Comment:
I would also assert that we have a valid callable before trying to use to materialize the `GBytes`: 

```suggestion
    if (!image->priv->bytes) {
        ASSERT(image->priv->createBytes);
        image->priv->bytes = image->priv->createBytes(image);
    }
    ASSERT(image->priv->bytes);
```